dsm/resource_acc_crypto_policy.go

```diff
@@ -0,0 +1,216 @@
+	return &schema.Resource{
+		CreateContext: resourceCreateAccountCryptoPolicy,
+		ReadContext:   resourceReadAccountCryptoPolicy,
+		UpdateContext: resourceCreateAccountCryptoPolicy,
+		DeleteContext: resourceDeleteAccountCryptoPolicy,
+		Schema: map[string]*schema.Schema{
+			"acct_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"approval_policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cryptographic_policy": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+// [C & U]: Create and Update Account Crypto Policy
+func resourceCreateAccountCryptoPolicy(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	cryptographic_policy := json.RawMessage(d.Get("cryptographic_policy").(string))
+
+	accountApprovalPolicyRead(ctx, d, m)
+
+	account_crypto_policy_object := make(map[string]interface{})
+	acct_id := d.Get("acct_id").(string)
+	operation := "PATCH"
+	url := fmt.Sprintf("sys/v1/accounts/%s", acct_id)
+
+	if _, ok := d.GetOk("approval_policy"); ok {
+		if debug_output {
+			tflog.Warn(ctx, "[C & U]: Approval policy is present.")
+		}
+		account_crypto_policy_object["method"] = "PATCH"
+		account_crypto_policy_object["operation"] = url
+		account_crypto_policy_object["body"] = map[string]interface{}{"cryptographic_policy": cryptographic_policy}
+		operation = "POST"
+		url = "sys/v1/approval_requests"
+	} else {
+		if debug_output {
+			tflog.Warn(ctx, "[C & U]: Approval policy is not set.")
+		}
+		account_crypto_policy_object["acct_id"] = acct_id
+		account_crypto_policy_object["cryptographic_policy"] = cryptographic_policy
+	}
+
+	resp, derr := m.(*api_client).APICallBody(operation, url, account_crypto_policy_object)
+	if derr != nil {
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "[DSM SDK] Unable to call DSM provider API client",
+			Detail:   fmt.Sprintf("[C & U]: API Call: %s %s: %v", operation, url, derr),
+		})
+		return diags
+	}
+
+	if debug_output {
+		resp_json, _ := json.Marshal(resp)
+		tflog.Warn(ctx, fmt.Sprintf("[C & U]: API response for cryptographic policy create operation: %s", resp_json))
+	}
+
+	d.SetId(acct_id)
+	return resourceReadAccountCryptoPolicy(ctx, d, m)
+}
+
+// [R]: Read Account Crypto Policy
+func resourceReadAccountCryptoPolicy(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	req, statuscode, err := m.(*api_client).APICall("GET", fmt.Sprintf("sys/v1/accounts/%s", d.Id()))
+	if statuscode == 404 {
+		d.SetId("")
+		diags = append(diags, diag.Diagnostic{
+			Severity: diag.Error,
+			Summary:  "[DSM SDK] DSM provider API client returned not found",
+			Detail:   fmt.Sprintf("[R]: API Call: GET sys/v1/accounts/%s", d.Id()),
+		})
+		return diags
+	} else {
```

Comment:
I'm pretty sure you can just pull the else block out, since the last statement of the if is a return statement